### PR TITLE
User Menu

### PR DESCRIPTION
**Pull request by Lukewh**
- Added user-menu component. 
- Updated header-help to use generic css. 
- Created generic css for header-menu css.
- Updated juju logo to redirect to /. 
- Updated html to arrange header elements.

Implements https://github.com/juju/juju-gui/issues/2572 with a tweak that the user name should never be shown for consistency - as discussed with @elvirafigueras.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7036/

File: jujugui/static/gui/src/app/components/user-menu/test-user-menu.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            className="header-menu__icon"
+            size="16" />
+        </span>
+        {userMenu.instance._generateUserMenu()}
+      </div>;
+    assert.deepEqual(userMenu.output, expected);
+  });
+
+  describe('menu', () => {
+    it('opens a menu when clicked', () => {
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/user-menu/test-user-menu.js]: Add a test for the click outside? Only if not too complicated.

File: jujugui/static/gui/src/app/components/user-menu/user-menu.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+/**
+  Provides a help menu to the header. The idea moving forward is to have a
+  more complete 'built-in' small help system for tips.
+*/
+YUI.add('user-menu', function() {
+
+  const UserMenu = React.createClass({
+
+    propTypes: {
+      LogoutLink: React.PropTypes.object,
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/components/user-menu/user-menu.js]: Yeah, had me banging my head against the desk.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7043/

File: jujugui/static/gui/src/app/app.js [hunk 3]
```diff
@@ -948,11 +948,11 @@ YUI.add('juju-gui', function(Y) {
       Renders the Log out component or log in link depending on the
       environment the GUI is executing in.
     */
-    _renderLoginOutLink: function() {
+    _renderUserMenu: function() {
       const controllerAPI = this.controllerAPI;
       const linkContainerId = 'profile-link-container';
       const linkContainer = document.getElementById(linkContainerId);
-      if (linkContainer === null) {
+      if (!linkContainer) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: Why was this change made? `getElementById` returns `null` if no element is found.

File: jujugui/static/gui/src/app/app.js [hunk 4]
```diff
@@ -993,8 +983,34 @@ YUI.add('juju-gui', function(Y) {
...
-        linkContainer);
+        />);
+
+      const navigateUserProfile = () => {
+        const auth = this._getAuth();
+        if (!auth) {
+          return;
+        }
+        views.utils.showProfile(
+         this.env && this.env.get('ecs'),
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: I think the indentation is off by 1 here.

File: jujugui/static/gui/src/app/app.js [hunk 5]
```diff
@@ -993,8 +983,34 @@ YUI.add('juju-gui', function(Y) {
...
+         this.state.changeState.bind(this.state), auth.rootUserName);
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: can you move the `auth.rootUserName` to a new line? I kept reading this as part of the `changeState` call :D

File: jujugui/static/gui/src/app/app.js [hunk 6]
```diff
@@ -993,8 +983,34 @@ YUI.add('juju-gui', function(Y) {
...
+      };
+
+      const navigateUserAccount = () => {
+        const auth = this._getAuth();
+        if (!auth) {
+          return;
+        }
+        views.utils.showAccount(
+         this.env && this.env.get('ecs'),
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: indentation

File: jujugui/static/gui/src/app/assets/css/_header.scss [hunk 7]
```diff
@@ -67,10 +67,140 @@
...
+    border-radius: 2px;
+    border-right-width: 0;
+    border: 1px solid $light-mid-grey;
+    transition: background-color .2s;
+    background-color: $canvas-background;
+    padding: 13px 11px;
+    margin: 12px 0 11px 12px;
+    // scss-lint:disable ImportantRule
+    color: $warm-grey !important;
+    // scss-lint:disable ImportantRule
```
Comment (hatched) [inline jujugui/static/gui/src/app/assets/css/_header.scss]: s/disable/enable ?

File: jujugui/static/gui/src/app/assets/css/_header.scss [hunk 8]
```diff
@@ -67,10 +67,140 @@
...
+        text-overflow: ellipsis;
+      }
+    }
+
+    &:hover {
+      background-color: $white;
+      // scss-lint:disable ImportantRule
+      color: $dark-warm-grey !important;
+      text-decoration: none !important;
+      // scss-lint:disable ImportantRule
```
Comment (hatched) [inline jujugui/static/gui/src/app/assets/css/_header.scss]: s/disable/enable ?

and the rest of them below ;)

File: jujugui/static/gui/src/app/components/user-menu/_user-menu.scss [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    margin-right: 10px;
+  }
+
+  .usso-login {
+    &__action {
+      text-decoration: none;
+      // Importants to overwrite vanilla values
+      // scss-lint:disable ImportantRule
+      color: $warm-grey !important;
+      // scss-lint:disable ImportantRule
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/user-menu/_user-menu.scss]: s/disable/enable :D

File: jujugui/static/gui/src/app/app.js (hunk 3 above)
Comment (kadams54) [inline jujugui/static/gui/src/app/app.js]: I actually like this change. Aside from being trivially shorter/less code, it's also more what I'm used to seeing in JS code. In my experience, asserting falsey is enough and the more specific assertion is only used if the case warrants it.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7057/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7058/

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**